packet_packer.go

```diff
@@ -185,9 +191,24 @@ func (p *packetPacker) composeNextPacket(
 	var payloadLength protocol.ByteCount
 	var payloadFrames []frames.Frame
 
+	// STOP_WAITING and ACK will always fit
 	if p.stopWaiting != nil {
-		p.controlFrames = append(p.controlFrames, p.stopWaiting)
+		payloadFrames = append(payloadFrames, p.stopWaiting)
+		l, err := p.stopWaiting.MinLength(p.version)
+		if err != nil {
+			return nil, err
+		}
+		payloadLength += l
 	}
+	if p.ackFrame != nil {
+		payloadFrames = append(payloadFrames, p.ackFrame)
+		l, err := p.ackFrame.MinLength(p.version)
```

Comment:
Yeah, I just don't like ignoring error values, as there's no compile-time check prohibiting you from e.g. changing AckFrame.MinLength to actually return an error.